codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -378,7 +382,6 @@ private void generateOperationDeserializer(GenerationContext context, OperationS
             }
             writer.write("");
 
-            writer.write("span.End()");
```

Comment:
is this expected?

Reply:
Yeah - for the deserialize middlewares, the initial defer is enough, because deserialize calls the rest of the middleware chain first.